src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MirrorEndpointsSpec.groovy

```diff
@@ -720,84 +644,62 @@ class MirrorEndpointsSpec extends HealthCheckSpecification {
+                    it.mirrorPointDirection = FlowPathDirection.FORWARD
+                    it.mirrorPointSwitchId = swPair.dst.dpId
+                    it.expectedError = new FlowMirrorPointNotCreatedExpectedError(~/The port $it.port on the switch \
 \'$swPair.dst.dpId\' is occupied by an ISL \(destination endpoint collision\)./)
                 }),
                 new MirrorErrorTestData("Unable to create a mirror point with s42Port conflict", {
                     def swPair = switchPairs.all().withDestinationSwitchConnectedToServer42().random()
-                    it.flow = flowHelperV2.randomFlow(swPair)
+                    it.flow = flowFactory.getBuilder(swPair).build()
                     def s42Port = swPair.dst.prop.server42Port
-                    it.mirrorPoint = FlowMirrorPointPayload.builder()
-                            .mirrorPointId(flowHelperV2.generateFlowId())
-                            .mirrorPointDirection(FlowPathDirection.FORWARD.toString())
-                            .mirrorPointSwitchId(swPair.dst.dpId)
-                            .sinkEndpoint(FlowEndpointV2.builder().switchId(swPair.dst.dpId)
-                                    .portNumber(s42Port)
-                                    .vlanId(flowHelperV2.randomVlan())
-                                    .build())
-                            .build()
+                    it.port = s42Port
+                    it.mirrorSinkEndpointSwitchId = flow.destination.switchId
+                    it.mirrorPointDirection = FlowPathDirection.FORWARD
+                    it.mirrorPointSwitchId = swPair.dst.dpId
                     it.expectedError = new FlowMirrorPointNotCreatedExpectedError(~/Server 42 port in the switch \
 properties for switch \'$swPair.dst.dpId\' is set to \'$s42Port\'. It is not possible to create or update an endpoint \
 with these parameters./)
                 })
         ].collectMany {
-            [it.tap { mirrorPoint.mirrorPointDirection = FlowPathDirection.FORWARD.toString() },
-             it.clone().tap { mirrorPoint.mirrorPointDirection = FlowPathDirection.REVERSE.toString() }]
+            [it.tap { mirrorPointDirection = FlowPathDirection.FORWARD },
+             it.clone().tap { mirrorPointDirection = FlowPathDirection.REVERSE }]
         }
     }
 
     @Tags([LOW_PRIORITY])
     def "Unable to create a mirror point with existing flow conflict, #mirrorDirection"() {
         given: "A flow"
         def swPair = switchPairs.all().random()
-        def flow = flowHelperV2.randomFlow(swPair)
-        def otherFlow = flowHelperV2.randomFlow(swPair, false, [flow])
-        flowHelperV2.addFlow(flow)
-        flowHelperV2.addFlow(otherFlow)
+        def flow = flowFactory.getRandom(swPair)
+        def busyEndpoints = []
+        busyEndpoints.addAll(flow.occupiedEndpoints())
+        def otherFlow = flowFactory.getBuilder(swPair, false, busyEndpoints).build().create()
```

Comment:
can be simplified: flowFactory.getBuilder(swPair, false, flow.occupiedEndpoints()).build().create()

Reply:
+